File: Basic_CPP/Variables_Functions_and_Scope/BeerOnTheWall.cpp

#include <sstream>
#include <iostream>
using namespace std;

void chorus( int n );
string beer( int n );
void take();

int main() {
	for( int i = 99; i >= 90; i-- )
		chorus( i );

}

void chorus( int n ) {
	cout << beer(n) << " on the wall, ";
	cout << beer(n) << endl;
	take();
	cout << beer( n-1 ) << " on the wall,"  << endl;
	
}

string beer( int n ) {
	string num;
	ostringstream t;
	t << n;
	return  t.str() + " bottles of beer";
}

void take() {
	cout << "Take one down and pass it around, ";
}

File: python/Python_LangFeatures/src/VariablesFunctionsAndScope/BeerOnTheWall.py
import sys
import wall
from sub import take
#the __init__.py in sub directory is mandatory

def main():
	for i in reversed(range(90,100)):
		chorus( i )

def chorus( n ):
	bottles( n )	
	wall.wall()
	bottles( n )
	print ""	
	take.take()
	bottles( n-1 )
	wall.wall()

def bottles( num ):
	sys.stdout.write( "{0} Bottles of beer ".format( num ) )

	
if __name__ == "__main__":
	main()


